core/src/test/scala/akka/persistence/r2dbc/internal/R2dbcExecutorSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.ActorSystem
+import akka.persistence.r2dbc.TestConfig
+import akka.persistence.r2dbc.TestData
+import akka.persistence.r2dbc.TestDbLifecycle
+import akka.persistence.r2dbc.internal.h2.H2Dialect
+import akka.persistence.r2dbc.internal.postgres.PostgresDialect
+import akka.persistence.r2dbc.internal.postgres.YugabyteDialect
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import io.r2dbc.spi.Connection
+import io.r2dbc.spi.R2dbcNonTransientResourceException
+import io.r2dbc.spi.Wrapped
+import org.scalatest.wordspec.AnyWordSpecLike
+
+object R2dbcExecutorSpec {
+  val config: Config = ConfigFactory
+    .parseString("""
+    akka.persistence.r2dbc.connection-factory {
+      initial-size = 1
+      max-size = 1
+      close-calls-exceeding = 3 seconds
+    }
+    """)
+    .withFallback(TestConfig.config)
+}
+
+class R2dbcExecutorSpec
+    extends ScalaTestWithActorTestKit(R2dbcExecutorSpec.config)
+    with AnyWordSpecLike
+    with TestDbLifecycle
+    with TestData
+    with LogCapturing {
+
+  override def typedSystem: ActorSystem[_] = system
+  private val table = "r2dbc_executor_spec"
+
+  case class Row(col: String)
+
+  // need pg_sleep or similar
+  private def canBeTestedWithDialect: Boolean =
+    r2dbcSettings.connectionFactorySettings.dialect == PostgresDialect ||
+    r2dbcSettings.connectionFactorySettings.dialect == YugabyteDialect
+
+  private def pendingIfCannotBeTestedWithDialect(): Unit = {
+    if (!canBeTestedWithDialect) {
+      info(s"Can't be tested with dialect [${r2dbcSettings.dialectName}]")
+      pending
```

Comment:
Annoying that there is no `ignore`, for things that aren't really pending to be fixed. (There is `cancel()` but that prints a stacktrace, which is worse than pending)